### PR TITLE
Upgradeable contracts

### PR DESCRIPTION
**Pull request by amandeeprehal-lab49**
Upgradeable contracts using a transparent proxy.

### CODE REVIEW

File: deployment/transparentUpgradeableProxy.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import {
+  AccountId
+} from "@hashgraph/sdk";
+import { Deployment } from "./deployContractOnTestnet";
+import ClientManagement from "../integrationTest/utils/utils";
+
+const clientManagement =  new ClientManagement();
+const {adminId} = clientManagement.getAdmin();
+
+//const swapContractAddress = "0x0000000000000000000000000000000002dc77c3";
```
Comment (mgarbs) [inline deployment/transparentUpgradeableProxy.ts:10]: Do we need this?

File: deployment/transparentUpgradeableProxy.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+const swapContractAddress = "0x0000000000000000000000000000000002de04e7";
+
+async function main() {
```
Comment (mgarbs) [inline deployment/transparentUpgradeableProxy.ts:14]: What is the benefit to using async await with promise chaining? Can't you accomplish the same thing without this chaining?


```
async main() {
     try {
        ...
     } catch (e) {
        // do something with error
     }
}
(async function() {
  await main();
})();
```

File: deployment/transparentUpgradeableProxy.ts (hunk 1 above)
Comment (amandeeprehal-lab49) [reply deployment/transparentUpgradeableProxy.ts:10]: It's an earlier version of Swap implementation. I just kept it for reference but not really required.

File: deployment/transparentUpgradeableProxy.ts (hunk 2 above)
Comment (amandeeprehal-lab49) [reply deployment/transparentUpgradeableProxy.ts:14]: I see your point but I find below one more cleaner as in the main we just need to care about logic and forget about error conditions. 

Another point, hardhat also follows the same pattern hence continued with it.

```
main()
  .then(() => process.exit(0))
  .catch((error) => {
    console.error(error);
    process.exit(1);
  });
```

File: deployment/transparentUpgradeableProxy.ts (hunk 1 above)
Comment (amandeeprehal-lab49) [reply deployment/transparentUpgradeableProxy.ts]: Removed.

**Review by damiano-melcarne-lab49 — APPROVED** [review]

**Pull request merged**